whelk-core/src/main/groovy/whelk/component/ElasticSearch.groovy

```diff
@@ -97,62 +93,54 @@ class ElasticSearch {
         }
     }
 
-    Tuple2<Integer, String> performRequest(String method, String path, String body, String contentType0 = null){
-
-        // Get an available connection from the pool
-        ConnectionPoolEntry httpConnectionEntry
-        int i = 0
-        while(true) {
-            i++
-            if (i == CONNECTION_POOL_SIZE) {
-                i = 0
-                Thread.yield()
-            }
-
-            synchronized (this) {
-                if (!httpConnectionPool[i].inUse.get()) {
-                    httpConnectionEntry = httpConnectionPool[i]
-                    httpConnectionEntry.inUse.set(true)
-                    break
-                }
-            }
+    Tuple2<Integer, String> performRequest(String method, String path, String body, String contentType0 = null) {
+        String host = elasticHosts[random.nextInt(elasticHosts.size())]
+        HttpRequest request
+        switch (method) {
+            case 'GET':
+                request = new HttpGet(host + path)
+                break
+            case 'PUT':
+                request = new HttpPut(host + path)
+                request.setEntity(httpEntity(body, contentType0))
+            case 'POST':
+                request = new HttpPost(host + path)
+                request.setEntity(httpEntity(body, contentType0))
+                break
+            default:
+                throw new WhelkRuntimeException("Bad request method:" + method)
         }
 
-        Tuple2<Integer, String> response = null
-
+        Tuple2<Integer, String> result = null
         try {
-            String contentType
-            if (contentType0 == null) {
-                contentType = ContentType.APPLICATION_JSON.toString()
-            } else {
-                contentType = contentType0
-            }
-
-            LongTermHttpConnection httpConnection = httpConnectionEntry.connection
             int backOffTime = 0
-            while (response == null || httpConnection.getResponseCode() == 429) {
+
```

Comment:
Perhaps use some of the available backoff/retry -strategies available in HttpClient instead?